src/main/java/com/google/api/codegen/ApiaryConfig.java

```diff
@@ -111,6 +111,20 @@
    */
   private String serviceVersion;
 
+  /*
+   * Supported auth types.
+   */
+  public enum AuthType {
+    API_KEY,
+    THREE_LEGGED_OAUTH,
```

Comment:
nit: I would probably put `OAUTH` in the front. And I might abbreviate `THREE_LEGGED`. Something like `OAUTH_3LO`. Though I realize the last `O` is redundant.... Up to you.